core/embed/rust/src/ui/model_mercury/component/instructions.rs

```diff
@@ -0,0 +1,125 @@
+use crate::{
+    strutil::TString,
+    ui::{
+        component::{text::TextStyle, Component, Event, EventCtx, Never},
+        constant::WIDTH,
+        geometry::{Alignment, Offset, Rect},
+        model_mercury::theme,
+        shape::{Renderer, Text},
+    },
+};
+
+/// Component showing a task instruction (e.g. "Swipe up") and optionally task
+/// description (e.g. "Confirm transaction") to a user. The component
+/// is typically placed at the bottom of the screen. The height of the provided
+/// area must be 18px (only instruction) or 37px (both description and
+/// instruction). The content and style of both description and instruction is
+/// configurable separatedly.
+pub struct Instructions<'a> {
+    area: Rect,
+    text_instruction: TString<'a>,
+    text_description: Option<TString<'a>>,
+    style_instruction: &'static TextStyle,
+    style_description: &'static TextStyle,
+}
+
+impl<'a> Instructions<'a> {
+    /// height for component with only instruction [px]
+    pub const HEIGHT_SIMPLE: i16 = 18;
+    /// height for component with both description and instruction [px]
+    pub const HEIGHT_DEFAULT: i16 = 37;
+
+    pub fn new<T: Into<TString<'a>>>(instruction: T) -> Self {
+        Self {
+            area: Rect::zero(),
+            text_instruction: instruction.into(),
+            text_description: None,
+            style_instruction: &theme::TEXT_SUB,
+            style_description: &theme::TEXT_SUB,
+        }
+    }
+
+    pub fn with_description<T: Into<TString<'a>>>(self, description: T) -> Self {
+        Self {
+            text_description: Some(description.into()),
+            ..self
+        }
+    }
+
+    pub fn update_instruction<T: Into<TString<'a>>>(&mut self, s: T) {
```

Comment:
Nit: usually our `update_...` functions take `ctx: &mut EventCtx` as an argument so they can call `.request_paint()` on it. Perhaps it's not going to be needed with the new graphics library but I'd do it here so that we're consistent across the whole model.

Reply:
fix: https://github.com/trezor/trezor-firmware/pull/3644/commits/494bb6e3dd885f6b603b2140ff6b636afbec62db